k8s/upgrades/0.8.0-0.8.1/cstor_volume_upgrade.sh

```diff
@@ -0,0 +1,162 @@
+
+################################################################# 
+# STEP: Generate deploy, replicaset and container names from PV #
+#                                                               #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc",  #
+#                                                               #
+# c-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-target        # 
+#################################################################
+
+c_dep=$(echo $pv-target)
+c_svc=$(echo $pv)
+c_vol=$(echo $pv)
+
+# Check if openebs resources exist and provisioned version is 0.8
+
+kubectl get deployment $c_dep -n $ns &>/dev/null
+rc=$?; if [ $rc -ne 0 ]; then echo "Target deployment not found: $rc"; exit; fi
+
+openebs_version=`kubectl get deployment $c_dep -n $ns -o jsonpath='{.metadata.labels.openebs\.io/version}'`
+if [ $openebs_version != "0.8.0" ]; then
+    echo "Current volumes version is not 0.8.0";exit 1;    
+fi
+
+kubectl get svc $c_svc -n $ns &>/dev/null
+rc=$?; if [ $rc -ne 0 ]; then echo "Service not found: $rc"; exit; fi
+
+
+# Fetch the older target and replica - ReplicaSet objects which need to be 
+# deleted before upgrading. If not deleted, the new pods will be stuck in 
+# creating state - due to affinity rules. 
+
+c_rs=$(kubectl get rs -o name --namespace $ns | grep $c_dep | cut -d '/' -f 2)
+
+
+################################################################ 
+# STEP: Update patch files with appropriate resource names     #
+#                                                              # 
+# NOTES: Placeholder for resourcename in the patch files are   #
+# replaced with respective values derived from the PV in the   #
+# previous step                                                #  
+################################################################
+
+sed "s/@sc_name/$sc_name/g" cstor-target-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-target-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-target-svc-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-target-svc-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-volume-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-volume-patch.json
+sed "s/@sc_name/$sc_name/g" cstor-volume-replica-patch.tpl.json | sed -u "s/@sc_resource_version/$sc_res_ver/g" > cstor-volume-replica-patch.json
+
+#################################################################################
+# STEP: Patch OpenEBS volume deployments (cstor-target, cstor-svc)              #  
+#################################################################################
+
+
+# #### PATCH TARGET DEPLOYMENT ####
+echo "Upgrading Target Deployment to 0.8.1"
+
+# Setting deployment startegy to recreate
+setDeploymentRecreateStrategy $ns $c_dep
+
+kubectl patch deployment  --namespace $ns $c_dep -p "$(cat cstor-target-patch.json)" 
+rc=$?; if [ $rc -ne 0 ]; then echo "Upgrade failed | ERROR: $rc"; exit; fi
```

Comment:
change message as "Upgrade failed for cstor-target-patch", and similarly, at other places

Reply:
ok